src/client/world.ts

```diff
@@ -129,6 +141,35 @@ class World {
     return this.getNeighbors(posB).map(coord => this.posIndex(coord)).includes(this.posIndex(posA));
   }
 
+  canBuildOn(tile: Tile): boolean {
+    return (
+      tile.owner?.civID === this.player.civID &&
+      tile.type !== 'ocean' &&
+      tile.type !== 'frozen_ocean' &&
+      tile.type !== 'mountain'
+    );
+  }
+
+  canSettleOn(tile: Tile): boolean {
+    return (
+      !tile.owner &&
+      tile.type !== 'ocean' &&
+      tile.type !== 'frozen_ocean' &&
+      tile.type !== 'mountain' &&
+      tile.type !== 'coastal' &&
+      tile.type !== 'frozen_coastal'
+    );
+  }
+
+  canFarmOn(tile: Tile): boolean {
+    // TODO - put this somewhere better
+    const farmableTiles = {
+      grass_lowlands: true,
+      plains: true,
+    };
+    return farmableTiles[tile.type];
```

Comment:
Shouldn't this be `return farmableTiles[tile.type] ?? false;`?
Otherwise, `canFarmOn` would return `undefined`. (Which *should* work fine in most cases, but it seems safer to make it always be a boolean).

Reply:
I think it gets automatically cast? Because TypeScript isn't throwing any sort of warning about it.